packages/block-library/src/post-excerpt/index.php

```diff
@@ -45,7 +29,39 @@ function render_block_core_post_excerpt( $attributes, $content, $block ) {
 	 * `excerpt_more` filter and return nothing. This will
 	 * result in showing only one `read more` link at a time.
 	 */
+	$more_text           = ! empty( $attributes['moreText'] ) ? '<a class="wp-block-post-excerpt__more-link" href="' . esc_url( get_the_permalink( $block->context['postId'] ) ) . '">' . wp_kses_post( $attributes['moreText'] ) . '</a>' : '';
+	$filter_excerpt_more = static function ( $more ) use ( $more_text ) {
+		// If the block has user input more text, return nothing as this is appended later.
+		$has_block_more_text = ! empty( $more_text );
+		if ( $has_block_more_text ) {
+			return '';
+		}
+		// If the default core more text is in use, replace it with the new block editor-y more text.
+		$default_more = ' ' . '[&hellip;]';
+		if ( $more === $default_more ) {
+			return '&hellip;';
```

Comment:
Should this be `$default_more`?

Here's what I'm seeing in this PR:

| Editor | Frontend |
|--------|--------|
| <img width="400" alt="Screenshot 2024-09-11 at 2 59 39 PM" src="https://github.com/user-attachments/assets/eb28ac8c-06fe-4a48-bcba-effacba365c0"> | <img width="400" alt="Screenshot 2024-09-11 at 2 59 44 PM" src="https://github.com/user-attachments/assets/819449ff-16ca-4da5-b98c-c8260951d8dc"> |

Reply:
Oh, I see it's replacing with `...`. 

Maybe we need to also filter `[...]` out of the editor?

Reply:
Interesting. It looks like this in the editor for me:
<img width="380" alt="Screenshot 2024-09-11 at 1 31 42 PM" src="https://github.com/user-attachments/assets/6366fe38-f9b9-4935-ba18-b60d875075c3">

I wonder how I can reproduce the `[...]` 🤔

Reply:
Oh, interesting. For me the `[...]` only kicks in with a large number of max words:

https://github.com/user-attachments/assets/69d8a609-97f8-4ddc-ae15-6a8002c6c4fe

Reply:
I think it might be coming from what the `rendered` version of the excerpt is from the API response. Here's what I'm seeing from inspecting the network request:

```
        "excerpt": {
            "raw": "",
            "rendered": "<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Duis maximus laoreet dolor, nec elementum turpis tincidunt eget. Donec dolor ligula, venenatis vel quam vulputate, consequat finibus dui. Proin posuere felis et maximus viverra. Nulla eu ipsum semper, pharetra augue pulvinar, tincidunt libero. Cras egestas tincidunt tortor non molestie. Nam purus dolor, tempor suscipit leo nec, [&hellip;]<\/p>\n",
            "protected": false
        },
```

So it's likely the server-rendered truncation?

Reply:
Is it related to the magic 55 default?

Reply:
Ah, that'd make sense. I wonder if the problem in the preview of the block is that it's using the response of the excerpt for the post itself, not the filtered length used for the rendering of this particular block... so the preview never gets a length of the excerpt longer than the 55? Just a hunch!

Here's a visual of what happens if we update the text in the post. The preview of the excerpt only updates in the editor UI once the API call happens:

https://github.com/user-attachments/assets/a5d7d385-0198-4c67-9a21-280898ff3a39

Reply:
_Are you both using the same theme to test?_
Since themes can also filter the "read more".

Reply:
I tested on a few different themes and I'm seeing `[...]` consistently > 55

Reply:
I was wondering why all this seemed familiar: 

- https://github.com/WordPress/gutenberg/pull/55029 😄 

> I wonder if the problem in the preview of the block is that it's using the response of the excerpt for the post itself, not the filtered length used for the rendering of this particular block... so the preview never gets a length of the excerpt longer than the 55? 

From memory, this sounds right. What's rendered to the page is the already-truncated excerpt, so increasing the characters in the editor doesn't do anything.

I'm just trying to remember where the `55` comes from. Not the block.json default. Here I changed it to 77:

https://github.com/user-attachments/assets/e59a5ee5-e710-4b4f-9636-a2e01b93250c

I suspect from the default filter when calling `get_the_excerpt()`

See: https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/default-filters.php#L215

Default value: https://github.com/WordPress/wordpress-develop/blob/8f90a7ebcd4d95a0b9f1aefd9f9ba4cf51d7d144/src/wp-includes/formatting.php#L3998

Reply:
I'm also seeing similar in trunk (but ever so slightly different), and another bug where the excerpt disappears completely when the post autosaves 😬 

Just trying to isolate what changes from this PR might be causing an issue (I haven't changed a lot with the editor implementation)

Reply:
I think the difference with the ellipsis is due to the `isTrimmed` logic here that I eagerly removed - https://github.com/WordPress/gutenberg/blob/4ca78cdf8ec71ba62dbf4403662202872d8e5888/packages/block-library/src/post-excerpt/edit.js#L187C8-L209

I've added it back. It'll still result in an issue where whenever trimming client side we'll have `&hellip;` but whenever trimming server side we'll have `[&hellip;]`. 🤔 

Also the issue with the excerpt being reduced to 55 words still remains (but it's the same now as in `trunk`). It seems like something this code is supposed to fix, but it doesn't work:
https://github.com/WordPress/gutenberg/blob/c80fc61804e99268eec4441e2915b33368c30b18/packages/block-library/src/post-excerpt/index.php#L84-L99

Reply:
I tracked the excerpt length bug down to that filter above, sometimes it's not being registered.

Looks like some aspects of the bug are being tracked in [the_excerpt() function return excerpt with different length in page load and ajax request](https://github.com/WordPress/gutenberg/issues/53570#top).

The idea in the future will be to use a param to define the excerpt length in the REST request. Looks like the REST API changes only merged to core in 6.6, but the plugin still supports 6.5, so that makes things tricky. We'd need to also replicate the REST API changes in the gutenberg codebase.

Reply:
> We'd need to also replicate the REST API changes in the gutenberg codebase.

Is that the first step do you think? Happy to help out here in any way.